src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/streaming_endpoint.py

```diff
@@ -73,6 +73,80 @@ def remove_akamai_access_control(client, account_name, resource_group_name, stre
     return client.update(resource_group_name, account_name, streaming_endpoint_name, streaming_endpoint)
 
 
+def update_streaming_endpoint_setter(client, resource_group_name, account_name, streaming_endpoint_name,
+                                     parameters):
+    if parameters.access_control is not None and parameters.access_control.ip is not None and parameters.access_control.ip.allow:  # pylint: disable=line-too-long
+        ips = list(map(lambda x: create_ip_range(streaming_endpoint_name, x) if isinstance(x, str) else x,
+                       parameters.access_control.ip.allow))
+        parameters.access_control.ip.allow = ips
+
+    return client.update(resource_group_name, account_name, streaming_endpoint_name, parameters)
+
+
+# pylint: disable=too-many-branches
+def update_streaming_endpoint(instance, tags=None, cross_domain_policy=None, client_access_policy=None,
+                              description=None, max_cache_age=None, ips=None, disable_cdn=None,
+                              cdn_provider=None, cdn_profile=None, custom_host_names=None):
+    from azure.mgmt.media.models import (IPAccessControl, StreamingEndpointAccessControl, CrossSiteAccessPolicies)
+
+    if not instance:
+        raise CLIError('The streaming endpoint resource was not found.')
+
+    if ips is not None:
+        is_ips_argument_empty = len(ips) == 1 and ips[0] == ""
+        if is_ips_argument_empty:
+            if instance.access_control is not None and instance.access_control.ip is not None:
+                instance.access_control.ip = None
+        else:
+            if instance.access_control is None:
+                instance.access_control = StreamingEndpointAccessControl()
+            if instance.access_control.ip is None:
+                instance.access_control.ip = IPAccessControl(allow=[])
+            for ip in ips:
+                instance.access_control.ip.allow.append(create_ip_range(instance.name, ip))
+
+    if instance.cross_site_access_policies is None:
+        instance.cross_site_access_policies = CrossSiteAccessPolicies()
+
+    if client_access_policy is not None:
+        if not client_access_policy:
```

Comment:
You are adding a double unnecessary IF statement. As far as I know None is a falsy value already. Just make a quick check to be sure, but given that's true, you don't need two ifs.

Reply:
Same applies for the following if statement in that file

Reply:
Same here.